Applications/spire/include/spire/book_view/quote_panel_indicator_widget.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef SPIRE_QUOTE_PANEL_INDICATOR_WIDGET_HPP
+#define SPIRE_QUOTE_PANEL_INDICATOR_WIDGET_HPP
+#include <chrono>
+#include <QTimer>
+#include <QWidget>
+
+namespace Spire {
+
+  //! Displays a BBO indicator panel with an animated background color.
+  class QuotePanelIndicatorWidget : public QWidget {
+    public:
+
+      //! Constructs a QuotePanelIndicatorWidget with a default background
+      //! color of black.
+      QuotePanelIndicatorWidget(QWidget* parent = nullptr);
+
+      //! Sets the background color.
+      /*
+        \param color The background color.
+      */
+      void set_color(const QColor& color);
+
+      //! Sets the background color with partial opacity and animates it
+      //! to full opacity.
+      /*
+        \param color The background color.
+      */
+      void animate_color(const QColor& color);
+
+    protected:
+      void paintEvent(QPaintEvent* event) override;
+
+    private:
+      static constexpr auto FADE_TIME_MS = 400;
```

Comment:
Use the same preference ranking you used for functions with constants as well.

Reply:
What do you mean about the preference ranking?

Reply:
The case you assigned to me about order of preference of functions applies to constants as well.